File: dotnet/cluster_management/GetCluster.cs
using Amazon;
using Amazon.DSQL;
using Amazon.DSQL.Model;
using Amazon.Runtime;

class GetCluster {
    public static async Task<GetClusterResponse> Get(RegionEndpoint region, string clusterId)
    {
        // Create the sdk client
        AWSCredentials aWSCredentials = FallbackCredentialsFactory.GetCredentials();
        AmazonDSQLConfig clientConfig = new()
        {
            AuthenticationServiceName = "dsql",
            RegionEndpoint = region
        };
        AmazonDSQLClient client = new(aWSCredentials, clientConfig);

        // Get cluster details
        GetClusterRequest getClusterRequest = new()
        {
            Identifier = clusterId
        };

        // Assert that operation is successful
        GetClusterResponse getClusterResponse = await client.GetClusterAsync(getClusterRequest);
        Console.WriteLine(getClusterResponse.Status);
        
        return getClusterResponse;
    }
}

File: cpp/cluster_management/src/GetCluster.cpp
#include "GetCluster.h"

using namespace Aws;
using namespace Aws::DSQL;
using namespace Aws::DSQL::Model;

ClusterStatus getCluster(const Aws::String& clusterId, Aws::DSQL::DSQLClient& client) {
    GetClusterRequest request;
    request.SetIdentifier(clusterId);
    GetClusterOutcome outcome = client.GetCluster(request);
    ClusterStatus status = ClusterStatus::NOT_SET;

    if (outcome.IsSuccess()) {
        const auto& cluster = outcome.GetResult();
        status = cluster.GetStatus();
    } else {
        std::cerr << "Get operation failed: " << outcome.GetError().GetMessage() << std::endl;
    }
    std::cout << "Cluster Status: " << ClusterStatusMapper::GetNameForClusterStatus(status) << std::endl;
    return status;
}

